zio-http-example/src/main/scala/example/WebSocketAdvanced.scala

```diff
@@ -42,11 +43,13 @@ object WebSocketAdvanced extends ZIOAppDefault {
       }
     }
 
-  val app: Http[Any, Nothing, Request, Response] =
-    Http.collectZIO[Request] {
-      case Method.GET -> Root / "greet" / name  => ZIO.succeed(Response.text(s"Greetings ${name}!"))
-      case Method.GET -> Root / "subscriptions" => socketApp.toResponse
-    }
+  val app: App[Any] =
+    Routes(
+      Method.GET / "greet" / string("name") -> handler { (name: String, req: Request) =>
+        Response.text(s"Greetings ${name}!")
+      },
+      Method.GET / "subscriptions"          -> handler(socketApp.toResponse),
```

Comment:
Although I like it this way, maybe we could have direct support to connect a route to a socket app (`-> socketApp`)?

Reply:
👍

Reply:
SocketApp erases to `Handler` which makes it more difficult to support this. I will look into it in a followup.